src/vtquery.cpp

```diff
@@ -352,6 +372,16 @@ struct Worker : Nan::AsyncWorker {
                     } // end tile.layer.feature loop
                 }     // end tile.layer loop
             }         // end tile loop
+            // Here we create "materialized" properties. We do this here because, when reading from a compressed
+            // buffer, it is unsafe to touch `feature.properties_vector` once we've left this loop.
+            // That is because the buffer may represent uncompressed data that is not in scope outside of Execute()
+            for (auto& feature : results_queue_) {
+                feature.properties_vector_materialized.reserve(feature.properties_vector.size());
+                for (auto const& property : feature.properties_vector) {
+                    auto val = vtzero::convert_property_value<mapbox::feature::value, mapbox::vector_tile::detail::property_value_mapping>(property.value());
+                    feature.properties_vector_materialized.emplace_back(std::string(property.key()), std::move(val));
+                }
+            }
         } catch (const std::exception& e) {
```

Comment:
[nitpick] Since `T const&` instead of  `const T&` is the style used throughout this code file, it might make sense to use this style for exceptions as well.

Reply:
Nice 👁 . Fixed in f98b2b6